src/H5private.h

```diff
@@ -2576,3 +2576,8 @@ H5_DLL herr_t H5_buffer_dump(FILE *stream, int indent, const uint8_t *buf, const
                              size_t buf_offset, size_t buf_size);
 
 #endif /* H5private_H */
+
+/* Check if a read of size bytes starting at ptr would overflow past
+ * the last valid byte, pointed to by buffer_end .
+ */
+#define H5_IS_BUFFER_OVERFLOW(ptr, size, buffer_end) (((ptr) + (size) - 1) > (buffer_end))
```

Comment:
This macro needs to be moved so that it's at least inside the big `#ifndef H5private_h .. #endif` block. For consistency, it should probably be located somewhere next to the other macros defined in this file.